web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -51,40 +48,104 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
   const Available = props.available;
-  const onSubmit = props.onSubmit;
 
   const doSearch = useCallback(
     async (
       event: FormEvent<HTMLFormElement> | undefined,
       { isInitialSubmit }: { isInitialSubmit: boolean }
     ): Promise<void> => {
       searchBusinessName(event)
-        .then(({ nameAvailability, submittedName }) => {
-          setNameAvailability(nameAvailability);
-          setSubmittedName(submittedName);
-          if (onSubmit) {
-            onSubmit(submittedName, nameAvailability, isInitialSubmit);
+        .then(async ({ nameAvailability, submittedName }) => {
+          if (props.isDba) {
+            if (!nameAvailability || !updateQueue) {
+              return;
+            }
+            setNameAvailability(nameAvailability);
+            if (nameAvailability.status === "AVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                  },
+                  dbaBusinessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  nexusDbaName: submittedName,
+                })
+                .update();
+            }
+          } else {
+            if (!nameAvailability || !updateQueue || isInitialSubmit) {
+              return;
+            }
+            setFieldsInteracted([FIELD_NAME]);
+            if (nameAvailability.status === "AVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                    businessName: submittedName,
+                  },
+                  businessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  businessName: submittedName,
+                  nexusDbaName: emptyProfileData.nexusDbaName,
+                  needsNexusDbaName: emptyProfileData.needsNexusDbaName,
+                })
+                .update();
+            } else if (nameAvailability.status === "UNAVAILABLE") {
+              await updateQueue
+                .queueFormationData({
+                  formationFormData: {
+                    ...updateQueue.current().formationData.formationFormData,
+                    businessName: submittedName,
+                  },
+                  businessNameAvailability: nameAvailability,
+                })
+                .queueProfileData({
+                  businessName: submittedName,
+                  needsNexusDbaName: true,
+                })
+                .update();
+            }
+
+            setFormationFormData((previousFormationData) => {
+              return {
+                ...previousFormationData,
+                businessName: submittedName,
+              };
+            });
```

Comment:
This is logic is ultimately going to live in `useBusinessNameSearch` when Business Name Search also uses this component. Was keeping this together for quick comparative purposes. Given that this is happening before the refactor is done, maybe it should be broken apart.